ext/bcmath/bcmath.c

```diff
@@ -198,11 +198,15 @@ static void php_str2num(bc_num *num, char *str)
 	char *p;
 
 	if (!(p = strchr(str, '.'))) {
-		bc_str2num(num, str, 0);
+		if (bc_str2num(num, str, 0)) {
```

Comment:
I find it somewhat confusing that this function returns 1 on *failure* and 0 on success. We should either flip this around, or use FAILURE/SUCCESS return values.

Reply:
I did it the "classic" way to allow for other error codes as well. Changing this is fine for me, though.